cmd/kube-controller-manager/app/core.go

```diff
@@ -282,12 +301,14 @@ func startResourceQuotaController(ctx ControllerContext) (bool, error) {
 
 	resourceQuotaController, err := resourcequotacontroller.NewResourceQuotaController(resourceQuotaControllerOptions)
 	if err != nil {
-		return false, err
+		return true, fmt.Errorf("Failed to start the resource quota controller: %v", err)
 	}
+
+	// Start the resourceQuota controller
 	go resourceQuotaController.Run(int(ctx.ComponentConfig.ResourceQuotaController.ConcurrentResourceQuotaSyncs), ctx.Stop)
 
-	// Periodically the quota controller to detect new resource types
-	go resourceQuotaController.Sync(discoveryFunc, 30*time.Second, ctx.Stop)
+	// Periodically sync the quota controller to detect new resource types
+	go resourceQuotaController.Sync(resourceQuotaControllerClient.Discovery(), 30*time.Second, ctx.Stop)
```

Comment:
is this intentional that `resourceQuotaControllerClient.Discovery()` is only executed once?